R/ExtractData_Soils.R

```diff
@@ -263,6 +263,192 @@ extract_soil_CONUSSOIL <- function(MMC, sim_size, sim_space, dir_ex_soil,
+  
+      if (!raster::compareCRS(sim_space[["crs_sites"]], soil_data)) {
+        sites_conus <- sp::spTransform(sites_conus, CRS = soil_data) #transform graphics::points to grid-coords
+      }
+  
+      if (sim_space[["scorp"]] == "point") {
+        cell_res_conus <- NULL
+        args_extract <- list(y = sites_conus, type = sim_space[["scorp"]])
+  
+      } else if (sim_space[["scorp"]] == "cell") {
+        cell_res_conus <- align_with_target_res(res_from = sim_space[["sim_res"]],
+                                                crs_from = sim_space[["sim_crs"]], sp = sim_space[["run_sites"]][todos, ],
+                                                crs_sp = sim_space[["crs_sites"]], crs_to = soil_data)
+        args_extract <- list(y = cell_res_conus, coords = sites_conus, method = "block",
+                             type = sim_space[["scorp"]])
+      }
+  
+      # actually extract the sand values from the sites
+      cond30 <- compiler::cmpfun(function(v) ifelse(is.na(v) | v < 30, NA, v))
+      ftemp <- tif_file
+      g <- if (file.exists(ftemp)) {
+        raster::brick(ftemp)
+      } else {
+        # bulk density of less than 0.3 g / cm3 should be treated as no soil
+        raster::calc(g, fun = cond30, filename = ftemp)
+      }
+      # get max depths for site soil profiles
+      ftemp_d <- paste0(dir.ex.gridded, "/depth_M_250m.tif");
+      gd <- if (file.exists(ftemp_d)) {
+        raster::brick(ftemp_d)
+      } else {
+        # bulk density of less than 0.3 g / cm3 should be treated as no soil
+        raster::calc(gd, fun = cond30, filename = ftemp_d)
+      }
+      soil_frame_depth <- do.call("extract_rSFSW2", args = c(args_extract, x = list(gd))) 
+      MMC[["data"]][todos, grep("depth", MMC[["cn"]])] = soil_frame_depth
+      if(soil_type != "depth"){
+        # get soil data as a dataframe 
+        soil <- do.call("extract_rSFSW2", args = c(args_extract, x = list(g))) 
+        soil_frame <- soil
+        percent_div <- 100
+  
+        # write density data, MMC[["data"]] requires a different column name to be written then the rest
+        if(soil_type == "matricd"){
+          percent_div <- 1000;
+          MMC[["data"]][todos, grep("density", MMC[["cn"]])[ils]][,soil_layer] <- soil_frame / percent_div 
+        }
+        else if(soil_type == "GravelContent"){
+          soil_frame <- pmax(pmin(soil_frame / 100, 1), 0)
+          # write gravel data to "data
+          MMC[["data"]][todos, grep("rock", MMC[["cn"]])[ils]][,soil_layer] <- soil_frame # gravel is already in form we want so no need to divide
+        }
+        else{
+          # write sand or clay data to "data
+          MMC[["data"]][todos, grep(soil_type, MMC[["cn"]])[ils]][,soil_layer] <- soil_frame / percent_div 
+        }
+  
+        # There is no organic carbon data, set all values to a default
+        MMC[["data"]][todos, grep("carbon", MMC[["cn"]])[ils]] <- default_TOC_GperKG
```

Comment:
Is this 'default_TOC_GperKG' necessary? None of the other functions seems to have to explicitily state that default_TOC_GperKG is 0.

Reply:
Conus does it the same way I do, where it is set as 0 as default in the function definition.